st2api/st2api/controllers/v1/actions.py

```diff
@@ -236,8 +237,17 @@ def delete(self, ref_or_id, requester_user):
             action_db,
         )
 
+        pack_name = action_db["pack"]
+        entry_point = action_db["entry_point"]
+        metadata_file = action_db["metadata_file"]
+
         try:
             Action.delete(action_db)
+            delete_action_files_from_pack(
+                pack_name=pack_name,
+                entry_point=entry_point,
+                metadata_file=metadata_file,
+            )
         except Exception as e:
```

Comment:
As the Action.delete and delete_action_files_from disk are in the same catch exception. If I get an error on deleting the files but not on the database, doesn't this mean that I'm still going to get a LOG.error saying "Database delete encountered exception"?

This would be a bit confusing. So either we need to amend the text slightly to indicate it might be database or file, or have two catch blocks.

Reply:
@mahesh-orch Please address this.

Reply:
Could you put the try/except blocks one after the other instead of nesting them?

Reply:
Thanks! that looks much better.